boot-qemu.py

```diff
@@ -661,16 +661,79 @@ def run(self):
         super().run()
 
 
+def guess_arch(kernel_arg):
+    # kernel_arg is either a path to the kernel build folder or a full kernel
+    # location. If it is a file, we need to strip off the basename so that we
+    # can easily navigate around with '..'.
+    if (kernel_dir := kernel_arg).is_file():
+        kernel_dir = kernel_dir.parent
+
+    # If kernel_location is the kernel build folder, vmlinux will be at
+    # <kernel_dir>/vmlinux
+    #
+    # If kernel_location is a full kernel location, it could either be:
+    #   * <kernel_dir>/vmlinux (if the image is vmlinux)
+    #   * <kernel_dir>/../../../vmlinux (if the image is in arch/*/boot/)
+    #
+    # Note: 'required=False' just to provide our own exception.
+    vmlinux_locations = ['vmlinux', '../../../vmlinux']
+    if not (vmlinux := utils.find_first_file(
+            kernel_dir, vmlinux_locations, required=False)):
+        raise RuntimeError(
+            'Architecture was not provided and vmlinux could not be found!')
+
+    if not (file := shutil.which('file')):
+        raise RuntimeError(
+            'Architecture was not provided and file is not installed!')
```

Comment:
```suggestion
            'Architecture was not provided and `file` is not installed!')
```

Just as a useful disambiguation, probably unnecessary.

Reply:
No, that is a good catch!

4e6963d09ecc5f74bc4d507248d352ee4a2606a9